backend/backend/src/main/java/de/aaaaaaah/velcom/backend/restapi/endpoints/GraphDetailEndpoint.java

```diff
@@ -120,6 +134,72 @@ public GetReply get(
 		return new GetReply(jsonDimensions, jsonGraphCommits);
 	}
 
+	/**
+	 * @param commits the commits to sort
+	 * @param hashes a map from hash to commit <em>for all commits in the commits list</em>
```

Comment:
This function has all the information to create this map itself. It doesn't need to be passed as argument.

Reply:
Also, this algorithm won't work correctly (leaf detection) if the hashes map contains additional commits not found in the commits list. As I understand it, the javadoc would still allow this.

Reply:
it could create the map itself, but the map is needed outside as well. I didn't want to calculate it twice.

Reply:
I missed that it was also used outside. Yeah, sounds better.